File: app/src/main/java/com/company/moviemania/Adapters/MovieListAdapter.java
package com.company.moviemania.Adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CenterCrop;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
import com.bumptech.glide.request.RequestOptions;
import com.company.moviemania.Activities.DetailActivity;
import com.company.moviemania.Domian.ListMovie;
import com.company.moviemania.R;

public class MovieListAdapter extends RecyclerView.Adapter<MovieListAdapter.ViewHolder> {

    ListMovie item;
    Context context;

    public MovieListAdapter(ListMovie item) {
        this.item = item;
    }

    @NonNull
    @Override
    public MovieListAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        View inflate = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_movies, parent, false);
        return new ViewHolder(inflate);
    }

    @Override
    public void onBindViewHolder(@NonNull MovieListAdapter.ViewHolder holder, int position) {
        holder.titleText.setText(item.getData().get(position).getTitle());
        RequestOptions requestOptions = new RequestOptions();
        requestOptions = requestOptions.transform(new CenterCrop(), new RoundedCorners(30));

        Glide.with(context)
                .load(item.getData().get(position).getPoster())
                .apply(requestOptions)
                .into(holder.banner);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(holder.itemView.getContext(), DetailActivity.class);
                i.putExtra("id", item.getData().get(position).getId());
                context.startActivity(i);
            }
        });
    }

    @Override
    public int getItemCount() {
        return item.getData().size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        TextView titleText;
        ImageView banner;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            titleText = itemView.findViewById(R.id.titleTxt);
            banner = itemView.findViewById(R.id.pic);
        }
    }
}

File: app/src/main/java/com/company/moviemania/adapters/MovieListAdapter.kt
package com.company.moviemania.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import com.company.moviemania.R
import com.company.moviemania.activities.DetailActivity
import com.company.moviemania.domain.ListMovie

class MovieListAdapter(var item: ListMovie) : RecyclerView.Adapter<MovieListAdapter.ViewHolder>() {
  var context: Context? = null

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
    context = parent.context
    val inflate = LayoutInflater.from(parent.context).inflate(R.layout.view_movies, parent, false)
    return ViewHolder(inflate)
  }

  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
    holder.titleText.text = item.data!![position]?.title
    var requestOptions = RequestOptions()
    requestOptions = requestOptions.transform(CenterCrop(), RoundedCorners(30))
    Glide.with(context!!)
      .load(item.data!![position].poster)
      .apply(requestOptions)
      .into(holder.banner)
    holder.itemView.setOnClickListener {
      val i = Intent(holder.itemView.context, DetailActivity::class.java)
      i.putExtra("id", item.data!![position].id)
      context!!.startActivity(i)
    }
  }

  override fun getItemCount(): Int {
    return item.data?.size!!
  }

  inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    var titleText: TextView
    var banner: ImageView

    init {
      titleText = itemView.findViewById(R.id.titleTxt)
      banner = itemView.findViewById(R.id.pic)
    }
  }
}
